diff --git a/Telegram/SourceFiles/core/launcher.cpp b/Telegram/SourceFiles/core/launcher.cpp
index 2765fe3ed..8f89a8410 100644
--- a/Telegram/SourceFiles/core/launcher.cpp
+++ b/Telegram/SourceFiles/core/launcher.cpp
@@ -61,6 +61,9 @@ FilteredCommandLineArguments::FilteredCommandLineArguments(
 		pushArgument(argv[i]);
 	}
 
+	pushArgument("-platformtheme");
+	pushArgument("xdgdesktopportal");
+
 #if defined Q_OS_WIN || defined Q_OS_MAC
 	if (OptionFreeType.value()) {
 		pushArgument("-platform");
