diff --git a/Telegram/SourceFiles/data/data_sponsored_messages.cpp b/Telegram/SourceFiles/data/data_sponsored_messages.cpp
index 66adb4d0f..52cf1d315 100644
--- a/Telegram/SourceFiles/data/data_sponsored_messages.cpp
+++ b/Telegram/SourceFiles/data/data_sponsored_messages.cpp
@@ -81,7 +81,8 @@ bool SponsoredMessages::append(not_null<History*> history) {
 }
 
 bool SponsoredMessages::canHaveFor(not_null<History*> history) const {
-	return history->peer->isChannel();
+	// return history->peer->isChannel();
+  return false;
 }
 
 void SponsoredMessages::request(not_null<History*> history) {
